### ISSUE TITLE
Missing lib with `brew install gdal --with-postgresql`

### ISSUE THREAD

**Issue by MatzFan** [Homebrew/homebrew-core#4899]
`brew install gdal --with-postgresql` exits OK, but one of the gdal executables is broken:
```
ogr2ogr --version
dyld: Library not loaded: /usr/local/lib/libjson-c.2.dylib
  Referenced from: /usr/local/bin/ogr2ogr
  Reason: image not found
Trace/BPT trap: 5
``` 

Reinstalling without postgresql fixes the problem:
```
brew uninstall gdal && brew install gdal
ogr2ogr --version
GDAL 1.11.5, released 2016/07/01
```
..but then I can't connect to Postgres databases.

Here is the [gist](https://gist.github.com/anonymous/d1e4e207d50671bc4e50bb149da1acd4) of the build - though looks like this ignored my `--with-postgresql` flag.

Postgresql is 9.5.4 (Homebrew installed) and reinstalling it after gdal install does not help.

A quick google shows this lib associated with Postgis [in at least one case](https://stackoverflow.com/questions/30901416/symbol-not-found-postgis-on-postgresql-9-3-3-mac-os-x-10-10). I did recently upgrade my databases from 9.3 to 9.5, so might this be the cause perhaps? Any help very greatly appreciated anyhow.

**Comment by zmwangx** [comment]
> though looks like this ignored my --with-postgresql flag.

It did not get ignored.

`--with-pg=/usr/local/bin/pg_config`

**Comment by MatzFan** [comment]
So I fixed this be reinstalling the `json-c` formula.

As I've been unable to recreate it in a CI environment (Travis) I'll assume it was something dodgy in my config. Closing the issue & leaving this record for posterity..

**Comment by apjanke** [comment]
Looks like `gdal` is missing a couple dependency declarations, including `json-c`. Should be fixed by #5554.
